frontend/src/Pages/Login.js

```diff
@@ -1,86 +1,179 @@
-			className="block h-8 w-8 mx-4 rounded-full overflow-hidden border-2 border-gray-300 hover:border-white"
-			type="button"
-			style={{ transition: "all .15s ease" }}
-		>
-			<img
-			alt="..."
-			className="h-full w-full mr-1"
-			src={imgUrl}
-			/>
-		</button>
-	)
-}
-
-AuthButton.propTypes = {
-	imgUrl: PropTypes.string
-}
-
-function UserField({ type }) {
-	switch (type) {
-		case 'Username': {
-			return (
-				<div className="w-full mb-3">
-					<input
-					type="username"
-					className="border-0 px-3 py-3 placeholder-gray-400 text-gray-700 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full"
-					placeholder="Username"
-					style={{ transition: "all .15s ease" }}
-					/>
-				</div>
-			)
-		}
+import UserField from '../Components/UserField'
+import SSOLogin from '../Components/SSOLogin'
+import {parsePhoneNumber} from "react-phone-number-input"
+import validateAndReturnUrl from '../HelperFunctions/validateAndReturnUrl'
+import axios from 'axios'
 
-		case 'Email address': {
-			return (
-				<div className="w-full mb-3">
-					<input
-					type="email"
-					className="border-0 px-3 py-3 placeholder-gray-400 text-gray-700 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full"
-					placeholder="Email"
-					style={{ transition: "all .15s ease" }}
-					/>
-				</div>
-			)
+
+export default function Login() {
+	const default_homeserver = 'matrix.org';
+
+	// Used to set the Identifier type for password based login
+	const [fieldType, setFieldType] = useState('Username');
+	const [userField, setUserField] = useState('');
+
+	// State variable to store the complete phone number including the country calling code with no '+'.
+	//Ex: 919676765xxxxx (India (91) )
+	const [phoneNumber, setPhoneNumber] = useState('');
+
+	const [password, setPassword] = useState('');
+
+	// Updated every time the homeserver input text is changed.
+	const [inputHomeServer, setInputHomeServer] = useState('matrix.org');
+
+	// Updated after save button is clicked
+	const [homeServer, setHomeServer] = useState(default_homeserver);
+
+	// List of available SSO providers
+	const [ssoProviders, setSSOProviders] = useState([]);
+	const [errorMessage, setErrorMessage] = useState('');
+
+	// Enable save button in homeserver when the inputHomeServer is being changed
+	// and Disable it when the save button is clicked.
+	const [disableSave, setDisableSave] = useState(true);
+
+	// Used to disable input fields when the homeserver url is invalid.
+	const [disableFields, setDisableFields] = useState(false);
+
+	const styleClassForFields = `border-0 px-3 py-3 placeholder-gray-400 text-gray-700 bg-white rounded text-sm shadow focus:outline-none focus:ring w-full disabled:opacity-50`;
+
+	// Fetches the available login types for a particular homeserver. Defaults to 'matrix.org'
+	useEffect(() => {
+		const fetchData = async () => {
+			const baseUrl = validateAndReturnUrl(homeServer);
+			const endpoint = "/_matrix/client/v3/login";
```

Comment:
It might be worth extracting these requests out to a MatrixClient class, so you can seperate the API calls from the code to some degree.